lib/junit.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module AutoHCK
+  class JUnit
+    def initialize(project)
+      @project = project
+      @logger = project.logger
+
+      @junit_template = ERB.new(File.read('lib/templates/junit.xml.erb'))
+    end
+
+    def run_property_drivers
+      return if @project.engine&.drivers.nil?
+
+      @project.engine.drivers.map(&:short).join(',')
+    end
+
+    def run_properties
+      {
+        'engine' => @project.engine_name,
+        'platform' => @project.engine_platform['name'],
+        'svvp' => @project.options.test.svvp,
+        'drivers' => run_property_drivers,
+        'hypervisor' => @project.setup_manager&.hypervisor_info,
+        'host' => @project.setup_manager&.host_info,
+        'results_url' => @project.result_uploader&.url,
+        'results_html' => @project.result_uploader&.html_url
+      }
+    end
+
+    def report_data
+      {
+        'tag' => @project.engine_tag,
+        'auto_hck_log' => @project.string_log.string.encode('ASCII', xml: :text),
```

Comment:
```suggestion
        'auto_hck_log' => @project.string_log.string.encode('UTF-8', xml: :text),
```

But I think it's better to fix the entire problem class by introducing Erubi.